.github/workflows/run-unit-tests.yml

```diff
@@ -89,17 +89,20 @@ jobs:
           docker context create $DOCKER_CONTEXT --docker host=unix:///var/run/docker.sock
 
           start_colima() {
-            # Find a free port
-            RANDOM_PORT=$((RANDOM % 16384 + 49152))
+            # Find a free port in the range 10000-20000
+            RANDOM_PORT=$((RANDOM % 10001 + 10000))
 
             echo "Using random port: $RANDOM_PORT for SSH"
 
-            colima start --network-address --arch x86_64 --cpu=1 --memory=1 --ssh-port $RANDOM_PORT
+            if ! colima start --network-address --arch x86_64 --cpu=1 --memory=1 --verbose --ssh-port $RANDOM_PORT; then
+              echo "Failed to start Colima."
+              return 1
+            fi
             docker context use $DOCKER_CONTEXT
           }
 
-          # Attempt to start Colima
-          ATTEMPT_LIMIT=3
+          # Attempt to start Colima; 4 retries:
```

Comment:
```suggestion
          # Attempt to start Colima; 5 retries:
```

Reply:
Read the code: 5 total attempts, which means 4 retries.